### DISCUSSION TITLE
Shelly Plug S switches off automatically - switches off in standby

### DISCUSSION THREAD

**Discussion by dippes** [arendst/Tasmota#22130] [Support request] [answered]
Can a Shelly Plug S be set so that the Shelly switches off when a connected consumer reaches a certain power threshold?

**Comment by sfromis** [comment]
You can use the `PowerHigh` command to set a threshold where Tasmota will generate extra messages (available to triggers in rules), when the energy consumption level steps above or below the threshold.

More generally, you can create rules triggering on any of the fields you see in the energy data in the published sensor payloads,

**Comment by dippes** [comment]
I tried that once

```
rule1 on energy#power1<10 do power1 0 endon

rule1 1

```

At < 10 watts the consumer should be switched off. But somehow it doesn't work.

Is the rule wrong?

**Reply by sfromis** [reply]
Yes, that's wrong, you need to match the field names from the JSON exactly. And there is no "Power1" there.

**Comment by dippes** [comment, marked as answer]
I changed the command.

When you type into the shell, Power turns off immediately.

But after switching on, <10 watts of power should switch off automatically, but it doesn't

`rule1 on energy#power<10 do power 0 endon`
``
`13:24:01.190 RSL: RESULT = {"Rule1":{"State":"ON","Once":"ON","StopOnError":"OFF","Length":35,"Free":476,"Rules":"on energy#power<10 do power 0 endon"}}
13:24:01.437 RUL: ENERGY#POWER<10 performs 'power 0'
13:24:01.442 RSL: RESULT = {"POWER":"OFF"}
13:24:01.445 RSL: POWER = OFF`

**Reply by dippes** [reply]
edit: It worked. Had to connect a consumer first.

**Comment by dippes** [comment]
can you add a timer to the rule?

Once the voltage drops below 10 watts, the Shelly switches off too quickly. (Not good for my Raspy)

There will be another 5 minute delay

**Comment by dippes** [comment]
Tried it that way, but it doesn't work.

The idea is to switch it off after a certain time if it is under 10 watts.

How does the rule become correct?
`rule1` on energy#power<10 do RuleTimer 60 do power 0 endon
`rule1` on energy#power<10 do PulseTime1 80 do power 0 endon

**Reply by sfromis** [reply]
Bad rules, you can only use `do` once per rule. To use a `RuleTimer`, you need an extra rule trigger `Rules#Timer` to match. The rules cookbook includes examples.

BTW. You may want to consider canceling the `RuleTimer` again if the consumption rises again before the timer expired.

**Comment by dippes** [comment]
I tried using Backlog.

The plug switches off after the first command even though a 100 W consumer is connected.

The second command is probably not recognized?

`14:04:56.539 RUL: Stored uncompressed, would compress from 41 to 33 (-20%)
14:04:56.543 RSL: RESULT = {"Rule1":{"State":"ON","Once":"ON","StopOnError":"OFF","Length":41,"Free":470,"Rules":"on energy#power<10 do Pulsetime1 60 ENDON"}}
14:04:56.758 RUL: Stored uncompressed, would compress from 24 to 20 (-17%)
14:04:56.761 RSL: RESULT = {"Rule2":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":24,"Free":487,"Rules":"on energy#power>10 ENDON"}}
14:04:57.075 RUL: ENERGY#POWER<10 performs 'Pulsetime1 60'
14:04:57.079 RSL: RESULT = {"PulseTime1":{"Set":60,"Remaining":60}}
14:05:03.152 RSL: RESULT = {"POWER":"OFF"}
14:05:03.155 RSL: POWER = OFF`

**Reply by sfromis** [reply]
Well, you need a valid rule. Every rule is of the form `on ... do ... endon`. Maybe also consider using one-shot mode.

And if you wanted to cancel the timer, you need to run a new `PulseTime` command, with the value 0.

**Reply by dippes** [reply]
Now it works as it should

The problem was that despite deleting the rules there was still a pulsetime of 60 in the system.

After a reset to 0 it works.

I would never have thought of that, I thought delete rules and everything is gone.
